rhp/v4/rhp_test.go

```diff
@@ -19,3 +20,212 @@ func TestMinRenterAllowance(t *testing.T) {
+	}
+	renterKey, hostKey := types.GeneratePrivateKey().PublicKey(), types.GeneratePrivateKey().PublicKey()
+
+	type testCase struct {
+		Description string
+		Modify      func(*types.V2FileContract, *RPCRenewContractParams)
+		RenterCost  types.Currency
+		HostCost    types.Currency
+	}
+
+	cases := []testCase{
+		{
+			Description: "empty",
+			Modify:      func(*types.V2FileContract, *RPCRenewContractParams) {},
+			RenterCost:  prices.ContractPrice,
+		},
+		{
+			Description: "no storage",
+			Modify: func(rev *types.V2FileContract, p *RPCRenewContractParams) {
+				p.Allowance = rev.RenterOutput.Value.Add(types.Siacoins(20))
+				p.Collateral = rev.TotalCollateral.Add(types.Siacoins(10))
+			},
+			RenterCost: types.Siacoins(20).Add(prices.ContractPrice),
+			HostCost:   types.Siacoins(10),
+		},
+		{
+			Description: "no storage - no renter rollover",
+			Modify: func(rev *types.V2FileContract, p *RPCRenewContractParams) {
+				p.Allowance = rev.RenterOutput.Value.Add(types.Siacoins(20))
+				p.Collateral = rev.TotalCollateral.Add(types.Siacoins(10))
+				// transfer all of the renter funds to the host so the renter will need to put up the entire allowance
+				rev.HostOutput.Value, rev.RenterOutput.Value = rev.HostOutput.Value.Add(rev.RenterOutput.Value), types.ZeroCurrency
+			},
+			RenterCost: types.Siacoins(320).Add(prices.ContractPrice),
+			HostCost:   types.Siacoins(10),
+		},
+		{
+			Description: "renewed storage - no additional funds",
+			Modify: func(rev *types.V2FileContract, p *RPCRenewContractParams) {
+				// add storage
+				rev.Capacity = SectorSize
+				rev.Filesize = SectorSize
+			},
+			RenterCost: prices.ContractPrice.Add(prices.StoragePrice.Mul64(SectorSize).Mul64(extension)), // storage cost is calculated for just the extension
+			HostCost:   types.ZeroCurrency,                                                               // collateral lock up is less than rollover
+		},
+		{
+			Description: "renewed storage",
+			Modify: func(rev *types.V2FileContract, p *RPCRenewContractParams) {
+				// add storage
+				rev.Capacity = SectorSize
+				rev.Filesize = SectorSize
+
+				// adjust the renewal params
+				p.Allowance = rev.RenterOutput.Value.Add(types.Siacoins(20))
+				p.Collateral = rev.TotalCollateral.Add(types.Siacoins(10))
+			},
+			RenterCost: types.Siacoins(20).Add(prices.ContractPrice).Add(prices.StoragePrice.Mul64(SectorSize).Mul64(extension)), // storage cost is calculated for just the extension
+			HostCost:   types.Siacoins(10).Add(prices.Collateral.Mul64(SectorSize).Mul64(renewalDuration)),                       // collateral is calculated for the full duration
+		},
+		{
+			Description: "renewed storage - no renter rollover",
+			Modify: func(rev *types.V2FileContract, p *RPCRenewContractParams) {
+				// adjust the renewal params
+				p.Allowance = rev.RenterOutput.Value.Add(types.Siacoins(20))
+				p.Collateral = rev.TotalCollateral.Add(types.Siacoins(10))
+
+				// add storage
```

Comment:
would the test be slightly better if we start out with a non-zero file size and effectively increase it here instead of setting it?